Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -743,4 +744,48 @@ Value ArrayPrototype::splice(Interpreter& interpreter)
     return removed_elements;
 }
 
+Value ArrayPrototype::fill(Interpreter& interpreter)
+{
+    auto *this_object = interpreter.this_value().to_object(interpreter);
+    if (!this_object)
+        return {};
+    
+    ssize_t length = get_length(interpreter, *this_object);
+    if (interpreter.exception())
+        return {};
+    
+    Value value;
+    if (interpreter.argument_count() >= 1)
+        value = interpreter.argument(0);
+    else
+        value = js_undefined();
+    
+    ssize_t relative_start = 0;
+    ssize_t relative_end = length;
+    
+    if (interpreter.argument_count() >= 2)
+        relative_start = interpreter.argument(1).to_i32(interpreter);
+    if (interpreter.argument_count() >= 3)
+        relative_end = interpreter.argument(2).to_i32(interpreter);
+    
+    size_t from, to;
+    
+    if (relative_start < 0)
+        from = max(length + relative_start, 0L);
+    else
+        from = min(relative_start, length);
+        
+    if (relative_end < 0)
+        to = max(length + relative_end, 0L);
+    else
+        to = min(relative_end, length);
+    
+    while (from < to) {
+        this_object->put_by_index(from, value);
+        from++;
+    }
```

Comment:
Missing exception check - also this could be written as:

```cpp
for (size_t i = from; i < to; ++i) {
    this_object->put_by_index(i, value);
    if (interpreter.exception())
        return {};
}
```